### DISCUSSION TITLE
using in place of global console

### DISCUSSION THREAD

**Discussion by mdagit** [adzejs/adze#178] [General]
I would like to replace the global console with something that calls into an adze logger, so as to intercept random other libs use of it, but when I do that i get an infinite loop because Log.print itself calls into console.

Note sure what to do about that.

**Comment by AJStacy** [comment]
Hey @mdagit,

Intercepting the standard console is generally bad practice. Adze itself is a fancy wrapper around the console to extend it. In order to capture the logs from other libraries, those libraries would need to provide a way to hook into their logs. Unfortunately, most do not do that.

If you absolutely must intercept these logs, I would suggest calling adze from your interceptor with the `silent` modifier (https://adzejs.com/reference/modifiers.html#silent) which will prevent the Adze logger from calling any standard console methods. You could then use a log listener (https://adzejs.com/reference/global-store.html#addlistener) or middleware (https://adzejs.com/reference/middleware.html) to capture the log message data and do something with it.

I hope this is helpful!

**Comment by mdagit** [comment]
Thanks that sounds like it would work. Another idea I had is that you could capture the function reference for console.log when your package is loaded, so that you aren't vulnerable to any later shenanigans. Or (equivalently) offer some configuration for "console_outputter" and the caller could supply something.

btw you are probably aware of this (and the reason you are using it), but the builtin console implementation has serialization capabilities that are not easily available to "user land" code.

**Reply by AJStacy** [reply]
That's an interesting idea to explore.
